python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -22,19 +43,23 @@ def get_s3_keys(
         raise botocore.exceptions.ClientError("Failed to initialize s3 client")
 
     paginator = s3_session.get_paginator("list_objects_v2")
-    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix)
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter="")
 
     contents = []
     for page in page_iterator:
         contents.extend(page.get("Contents", []))
```

Comment:
Another observation, this could likely be made more efficient by leveraging the default sorting order:
<img width="822" alt="image" src="https://github.com/user-attachments/assets/2fef68da-995b-4b34-bf32-db6f13e4cf61">

But unfortunately it's different depending on the type of bucket.